src/gui/DatabaseTabWidget.cpp

```diff
@@ -49,7 +49,7 @@ DatabaseTabWidget::DatabaseTabWidget(QWidget* parent)
     : QTabWidget(parent)
     , m_dbWidgetStateSync(new DatabaseWidgetStateSync(this))
     , m_dbWidgetPendingLock(nullptr)
-    , m_databaseOpenDialog(new DatabaseOpenDialog())
+    , m_databaseOpenDialog(new DatabaseOpenDialog(this))
```

Comment:
Why are you changing this?

Reply:
Because otherwise, the testing code has no way to access the database open dialog. And changing from using a `QScopedPointer` to using `QObject` parent should not change the lifetime of `DatabaseOpenDialog` in any non-negligible way.

Reply:
Don't change the ABI for testing purposes. We have that in other places and I removed a bunch of it for 2.6. You can get the current active window (the dialog in this case) with `qApp->activeWindow()`.

Reply:
Sorry I was wrong with this one, you are correct the ABI did not change.